src/test/suite/index.ts

```diff
@@ -29,52 +25,54 @@ export function run(): Promise<void> {
   const testsRoot = path.join(__dirname, '..');
 
   return new Promise((c, e) => {
```

Comment:
I'm not really sure why we use `new Promise` with that chain of `then` and `catch` instead of making `run` an async function.

If callbacks is the issue we can just import `util.promisify` and turn those into something that can be awaited. 

```
const globAsync = promisify(glob);
```

or 

```
const runMocha = promisify(mocha.run.bind(mocha));
```

This mixture of async code styles is something i've noticed frequently in VSCode, is so much easier to have `promisify` from the node standard library  and `async/await`  to take care of all the exception handling with all the silly corner cases. It makes the code a lot more readable, safe and easy to maintain.